### ISSUE TITLE
Odd timeout errors when connecting to echo.webtransport.day on Windows

### ISSUE THREAD

**Issue by MOZGIII** [BiagioFesta/wtransport#33]
Hello! I have an odd error in my project. I'm getting a `ConnectionError(TimedOut)` when connecting to `https://echo.webtransport.day`, but only on Windows.
Here's a CI run: https://github.com/MOZGIII/xwebtransport/actions/runs/5629426054/job/15254370622

You should be able to reproduce by simply doing `cargo test` at https://github.com/MOZGIII/xwebtransport on Windows.

**Comment by BiagioFesta** [comment]
The problem is related to the IP configuration of your client/test.

In the failing test, you are binding the client IP to the `INADDR_ANY` **IPv6**. [See here](https://github.com/MOZGIII/xwebtransport/blob/86f7deccdf6a33679728ab93d671cabdf3a395e4/crates/xwebtransport-wtransport/tests/test.rs#L7): `"[::]:0"`.

On Windows, *dual-stack* sockets is disabled by default. It means that client bound to IPv6 address will not able to connect to an IPv4 address.

In short, you have a IPv6 client trying to connect to a IPv4 (the host you are connecting to in your test resolves into a IPv4 address).

You should be able to solve the issue just binding the client to IPv4, that is, `0.0.0.0:0`

**Comment by BiagioFesta** [comment]
We might want to consider to add documentation to the `Endpoint<Client>::connect` method regarding dual-stack networking, as [quinn does](https://docs.rs/quinn/latest/quinn/struct.Endpoint.html#method.client)

**Comment by MOZGIII** [comment]
Oh, I wasn't aware of that. Good to know, thank you.

It brings me the other point: the configurability is very limited currently. It would be nice to be able to do ioctls on the socket, or bring your own socket, etc. I'll create another issue for that.
